src/app/store/ProductList/ProductList.reducer.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+import {
+    APPEND_PAGE,
+    UPDATE_PRODUCT_LIST_ITEMS,
+    UPDATE_LOAD_STATUS
+} from 'Store/ProductList';
+
+const initialState = {
+    pages: {},
+    isLoading: true
+};
+
+const ProductListReducer = (state = initialState, action) => {
+    const {
+        items,
+        currentPage,
+        isLoading
+    } = action;
+
+    if (items) {
```

Comment:
We should someday get rid of this :) But not in this PR

Reply:
I will look what we can do with it soon.